js/style/style_layer/line_style_layer.js

```diff
@@ -11,18 +11,20 @@ module.exports = LineStyleLayer;
 
 LineStyleLayer.prototype = util.inherit(StyleLayer, {
 
-    getPaintValue: function(name, zoom) {
-        var output = StyleLayer.prototype.getPaintValue.apply(this, arguments);
+    recalculate: function(zoom, zoomHistory) {
+        StyleLayer.prototype.recalculate.call(this, zoom, zoomHistory);
 
         // If the line is dashed, scale the dash lengths by the line
         // width at the previous round zoom level.
-        if (output && name === 'line-dasharray') {
-            var lineWidth = this.getPaintValue('line-width', Math.floor(zoom), Infinity);
-            output.fromScale *= lineWidth;
-            output.toScale *= lineWidth;
+        if (this.paint['line-dasharray']) {
+            var flooredZoom = Math.floor(zoom);
+            if (this._flooredZoom !== flooredZoom) {
+                this._flooredZoom = flooredZoom;
+                this._flooredLineWidth = this.getPaintValue('line-width', flooredZoom, Infinity);
+            }
+
+            this.paint['line-dasharray'].fromScale *= this._flooredLineWidth;
+            this.paint['line-dasharray'].toScale *= this._flooredLineWidth;
         }
-
-        return output;
     }
```

Comment:
Right, moved it back! Turned not to have much impact since additional `getPaintValue` only gets called when a round zoom is changed.